sdl_viewer/src/main.rs

```diff
@@ -210,57 +211,66 @@ impl NodeView {
 // Keeps track of the nodes that were requested in-order and loads then one by one on request.
 struct NodeViewContainer {
     node_views: HashMap<octree::NodeId, NodeView>,
-    queue: VecDeque<octree::NodeId>,
-    queued: HashSet<octree::NodeId>,
+    requested: HashSet<octree::NodeId>,
+    node_id_sender: Sender<octree::NodeId>,
+    node_data_receiver: Receiver<(octree::NodeId, octree::NodeData)>,
 }
 
 impl NodeViewContainer {
-    fn new() -> Self {
+    fn new(octree: Arc<octree::Octree>) -> Self {
+        let (node_id_sender, node_id_receiver) = mpsc::channel();
```

Comment:
Add comment, something along the lines of: "We do the expensive IO in a background thread and communicate with it through channels".